### PR TITLE
Add gitea manager reload-templates command

### PR DESCRIPTION
**Pull request by brechtvl**
This can be useful to update custom templates in production mode, when they are updated frequently and a full Gitea restart each time is disruptive.

### CODE REVIEW

**Comment by brechtvl** [comment]
This might be a bit obscure, I'm not sure how generally useful this functionality is. But our templates display things like release status and ongoing projects, and it's nice to be able to update those without risking e.g. data loss when submitting comments or failed git repository access to update those.

**Comment by lunny** [comment]
Maybe we also need a template load test sub command to do a test before real reload.

**Review by lunny — APPROVED** [review]

File: modules/templates/htmlrenderer.go [hunk 1]
```diff
@@ -96,6 +99,16 @@ func HTMLRenderer() *HTMLRender {
...
 }
 
+func ReloadHTMLTemplates(test bool) error {
+	if err := htmlRender.CompileTemplates(test); err != nil {
+		if !test {
+			log.Error("Template error: %v\n%s", err, log.Stack(2))
+		}
+		return err
+	}
+	return nil
```
Comment (wxiaoguang) [inline modules/templates/htmlrenderer.go]: A nit (not block) I think the code could be as simplified to `return htmlRender.CompileTemplates(test)`, let caller decide whether print the error?

----

Update: does it really need "test"? comment below ....

**Comment by wxiaoguang** [comment]
> Maybe we also need a template load test sub command to do a test before real reload.

I see the original PR doesn't have so, and I do not think there is a real case which really requires "testing".  If there is any error, the old templates are still used.

Do you think this is a must?  If no, I would suggest to keep things simple. @lunny

**Comment by lunny** [comment]
> > Maybe we also need a template load test sub command to do a test before real reload.
> 
> I see the original PR doesn't have so, and I do not think there is a real case which really requires "testing". If there is any error, the old templates are still used.
> 
> Do you think this is a must? If no, I would suggest to keep things simple (no test). @lunny

If the reload failure will not break the old templates I agree we don't need the test.

**Comment by brechtvl** [comment]
I removed the `--test` flag now. When there is an error on reload, the old templates are kept. It was already implemented to work this way for auto reloading in dev mode.

**Review by wxiaoguang — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**